### ISSUE TITLE
Any way to use this without expo?

### ISSUE THREAD

**Issue by shokimble** [expo/expo-three#43]
I really like what has been done to get three.js working properly on react native and appreciate that this project is literally about three.js on expo but I don't use expo and don't want to add 30 odd megabytes to my app size just to get WebGL and three.js working.

I tried installing it and even though expo isn't in the dependency list (it should be as it stands) it fails to work as it needs expo.

Trawling through the code I see most of the references are to small expo utility functions so it seems like the dependency could be unwound pretty easily.

Is there any interest in doing this from the maintainer(s) or is there a want for business reasons of platform lock in?

**Comment by EvanBacon** [comment]
TL;DR: No
--

First it's important to know that Expo is a layer of APIs that are built and maintained by core React Native contributors, not a competing product. 
So I have no agenda to lock anyone in - hence agnostic 3D GL library.

The way WebGL works in React Native is pretty complex and it uses a library called EXGL. 
EXLG is a C++ library that is integrated into Expo and has a corresponding Java, and Objective-C interface. 
With this we can bind the WebGL context to OpenGL-ES. Three.js then wraps WebGL to make it easier to work with. There have been efforts to extract this library from Expo but they have fallen out of date, and lack a lot of features that EXGL has (like WebGL 2.0 (way faster 😊) ). 

Some people at Expo are currently working on ways to make libraries more abstract so they can remove modules that you don't need. Currently this is somewhat implemented with ARKit, but this has downsides as you must push new native code to your builds. 

I currently have 3 EXGL apps in the store and they all do pretty well. Haven't heard complaints about app size since most people who play 3D video games on mobile don't really pay attention to size. 😅 (which of course is reduced when compiled) 🦄

**Comment by shokimble** [comment]
Hey appreciate that Even. I didn't realise it was so complex. On the surface it seemed quite trivial. Guess I didn't dig deep enough.

There is a project that split off the core functionality of the webgl side which is what I have been using but it has major issues with loading in textures which is pretty much a deal breaker when you're trying to load in models and is quite far behind Expo.

https://github.com/react-community/react-native-webgl

I do like Expo and appreciate what you guys have done but it's very monolithic as mentioned so it's either a choice you make early on and embrace the ecosystem but bolting it onto a mature app can be a world of pain.

**Comment by skizzo** [comment]
Hi,
what's the status of this in 2020? Are there some docs that explain how to use `expo-three` without Expo itself? Thanks for some infos :)

**Comment by ToyboxZach** [comment]
Same question, the links in this issue are a bunch of deprecated packages and there doesn't seem to be any info for the last couple years I can find

**Comment by 3GOMESz** [comment]
2021. Things are going well with [unimodules](https://docs.expo.io/bare/installing-unimodules/). Except for this https://github.com/expo/expo-three/issues/196

**Comment by ToyboxZach** [comment]
I get stuck with 
TypeError: null is not an object (evaluating 'NativeUnimoduleProxy.viewManagersNames') 

And there isn't any good documentation out there about what that means.
